lib/howl/buffer.moon

```diff
@@ -277,6 +278,19 @@ class Buffer extends PropertyObject
       b_end_pos = @byte_offset end_pos
       @_buffer\ensure_styled_to pos: b_end_pos
 
+  mode_at: (pos) =>
+    b_pos = @byte_offset pos
+    mode_name = @_buffer.styling\get_mode_name_at b_pos
+    -- Returns @mode if there's no marker or the requested mode doesn't exist.
+    mode_name and mode.by_name(mode_name) or @mode
+
+  config_at: (pos) =>
+    mode_at = @mode_at pos
+    return @config if mode_at == @mode
+    new_config = config.local_proxy!
```

Comment:
We could change the `word_pattern` to a mode field instead, but there are still other settings that at least should be used (e.g. indentation, tab width, etc).